### PR TITLE
FAQ: small improvement in 'this type' answer

### PR DESCRIPTION
**Pull request by SethTisue**
This comes up all the time, most recently at https://users.scala-lang.org/t/guarantees-of-this-type/7822

### CODE REVIEW

**Review by BalmungSan — COMMENTED** [review]
I would rephrase it a bit and maybe move it below the current paragraphs as a _"note"_.

> **Note:** Using `this.type` won't work; **unless your method will always return `this`**.
> People often try that **to refer to the same type of the current instance**, but `this.type` means "the singleton type of this instance", a different and too-specific meaning.

_(bold to emphasize my modifications)_

**Comment by SethTisue** [comment]
I tweaked it a bit but I like keeping `this.type` first. 🤷, nbd either way I guess
